docs/schema_design.md

```diff
@@ -3,52 +3,41 @@
 ## Sql Schema(postgresql)
 ```sql
 CREATE TABLE blocks (
-    number NUMERIC PRIMARY KEY,
-    hash TEXT UNIQUE NOT NULL,
-    parent_hash TEXT NOT NULL,
-    logs_bloom TEXT NOT NULL,
-    gas_limit NUMERIC NOT NULL,
-    gas_used NUMERIC NOT NULL,
-    timestamp TIMESTAMPTZ NOT NULL,
-    miner TEXT NOT NULL,
-    size NUMERIC NOT NULL
-    -- extra_data TEXT
-    -- sha3Uncles TEXT,
-    -- uncles []TEXT,
-    -- state_root TEXT,
-    -- transactions_root TEXT,
-    -- receipts_root TEXT,
-    -- difficulty NUMERIC,
-    -- total_difficulty NUMERIC,
-    -- nonce TEXT
+    number numeric PRIMARY KEY,
+    hash bytea NOT NULL,
+    parent_hash bytea NOT NULL,
+    gas_limit numeric NOT NULL,
+    gas_used numeric NOT NULL,
+    miner bytea NOT NULL,
+    size integer NOT NULL,
+    "timestamp" timestamp with time zone NOT NULL
 );
 
+create unique index on blocks(hash);
+
 CREATE TABLE transactions (
     id BIGSERIAL PRIMARY KEY,
-    hash TEXT UNIQUE NOT NULL,
-    eth_tx_hash TEXT UNIQUE NOT NULL,
-    block_number NUMERIC REFERENCES blocks(number) NOT NULL,
-    block_hash TEXT NOT NULL,
-    transaction_index INTEGER NOT NULL,
-    from_address TEXT NOT NULL,
-    to_address TEXT,
-    value NUMERIC NOT NULL,
-    nonce NUMERIC,
-    gas_limit NUMERIC,
-    gas_price NUMERIC,
-    input TEXT,
-    v NUMERIC NOT NULL,
-    r TEXT NOT NULL,
-    s TEXT NOT NULL,
-    cumulative_gas_used NUMERIC,
-    gas_used NUMERIC,
-    logs_bloom TEXT NOT NULL,
-    contract_address TEXT,
-    status BOOLEAN NOT NULL
+    hash bytea NOT NULL,
+    eth_tx_hash bytea NOT NULL,
+    block_number numeric REFERENCES blocks(number) NOT NULL,,
+    block_hash bytea NOT NULL,
+    transaction_index integer NOT NULL,
+    from_address bytea NOT NULL,
+    to_address bytea,
+    value numeric(80,0) NOT NULL,
+    nonce bigint NOT NULL,
+    gas_limit numeric,
+    gas_price numeric,
+    input bytea,
+    v smallint NOT NULL,
```

Comment:
is the v stored here in database not containing chain_id ?

Reply:
`v` here is just last byte of signature, `r + s + v` is a 65-byte signature

https://github.com/nervosnetwork/godwoken-web3/blob/refactor-db/crates/indexer/src/indexer.rs#L236

Reply:
ok, I forgot. the signature's V is either 27 or 28 without chain_id. only the signing message is hashed from tx with V including chain_id.